pkg/validator/app.go

```diff
@@ -463,6 +463,25 @@ func admitAppCR(
 			// to null. See the commit comments in the PR that closes issue
 			// #319 for more details.
 			prevAppCR.Spec.DefaultSetupPackage = appCR.Spec.DefaultSetupPackage
+
+			// EZML-862
+			// Before doing the comparison, we should ignore defferences
+			// between the images defined for different roles. The app configuration
+			// should be able to replace if the image is changed
+			// even when some KD cluster instance based on this spec is alive.
+			if appCR.Spec.DefaultImageRepoTag != nil {
+				*prevAppCR.Spec.DefaultImageRepoTag = *appCR.Spec.DefaultImageRepoTag
+			}
```

Comment:
I'm dubious about whether we should allow setting the default image for an existing kdapp. The semantics of default image are:

At create time, if any role does not specify an image then it will get the default image. The default image is then removed from the spec.

So if we modify default image for an existing kdapp it will never have any effect, since all roles will have an image by that time. That's probably counter to what folks expect.

Reply:
Removed